src/spikeinterface/core/numpyextractors.py

```diff
@@ -99,6 +116,115 @@ def get_traces(self, start_frame, end_frame, channel_indices):
             traces = traces[:, channel_indices]
 
         return traces
+    
+
+
+class SharedMemoryRecording(BaseRecording):
+    """
+    In memory recording with shared memmory buffer.
+    
+
+    Parameters
+    ----------
+    shm_names: list
+        List of sharedmem names.
+    shape_list: list
+
+    sampling_frequency: float
+        The sampling frequency in Hz
+    t_starts: None or list of float
+        Times in seconds of the first sample for each segment
+    channel_ids: list
+        An optional list of channel_ids. If None, linear channels are assumed
+    """
```

Comment:
```suggestion
    channel_ids: list, default: None
        An optional list of channel_ids. If None, linear channels are assumed
    t_starts: None or list of float, default: None
        Times in seconds of the first sample for each segment
    main_shm_owner: bool, default: True
    
    """
```

add in shape_list def and `main_shm_owner` def

Reply:
done :)